### ISSUE TITLE
Add `on_error` argument to linalg functions

### ISSUE THREAD

**Issue by jessegrabowski** [pymc-devs/pytensor#1078]
### Description

Our implementation of `Cholesky` has a nice argument `on_error` that lets the user get back `nan` from an bad input instead of halting the program. This would be a nice feature to have in other places where shoving in random matrices (pun intended) can be dangerous. I'm thinking specifically about `pt.linalg.solve`, but I kept the issue name generic in case we think of others.

The fix itself is trivial: add a try/except around the `perform` method of e.g. solve. The only wrinkle I foresee is what to do with the gradients.

**Comment by ricardoV94** [comment]
> The only wrinkle I foresee is what to do with the gradients.

Do we do anything on the Solve?

**Comment by jessegrabowski** [comment]
On CholeskySolve you mean? Or just solve generally?

Right now if `pt.linalg.solve` fails (for example because you pass a low-rank matrix) the program crashes with a LAPACK error

**Comment by ricardoV94** [comment]
Sorry I meant on the Cholesky that has this special behavior implemented. Do we do anything different on the gradient then if we didn't?

**Comment by jessegrabowski** [comment]
Yeah it [gets replaced with a scalar 1 ](https://github.com/pymc-devs/pytensor/blob/f2ad711ff22c4d56d210e4bd67c1e727481fd818/pytensor/tensor/slinalg.py#L109)(which I guess implies the node has no contribution to the gradients?)

**Comment by ricardoV94** [comment]
So it returns `nan`, and it has the work-around for Solve. If Solve returned nan, it wouldn't need the work-around?

**Comment by ricardoV94** [comment]
This would also allow us to remove this switch logic in the multivariate logp: https://github.com/pymc-devs/pymc/blob/f5517186874107e72440ec9b55cd31ce77cc2a3d/pymc/distributions/multivariate.py#L181-L183
